gslab_scons/log.py

```diff
@@ -78,7 +82,11 @@ def collect_builder_logs(parent_dir):
                 with open(f_full, 'rU') as g:
                     s = g.readlines()[1] # skip a line
                     s = s[s.find('{')+1: s.find('}')] # find {} time identifier 
-                    builder_log_end_time = datetime.strptime(s, "%Y-%m-%d %H:%M:%S")
-                builder_log_collect[f_full] = builder_log_end_time
+                    try:
+                        builder_log_end_time = datetime.strptime(s, "%Y-%m-%d %H:%M:%S")
+                    except ValueError: # if the code breaks, there's no time identifier
+                        beginning_of_time    = datetime.min
+                        builder_log_end_time = beginning_of_time
+                builder_log_collect[f_full]  = builder_log_end_time
```

Comment:
It's a way to catch broken logs. So there are two behaviors associated with your code failing. 

1) Your code fails within the process. The process still finish, but your code is broken. In that case the `atexit` logging functionality works: your log has two timestamps, and `scons` think the action completed properly while in reality it hasn't. [Here is an example.](https://gist.github.com/stanfordquan/5cf8d25f8e23f18a513f8bd040f6fe3e) You see that Scons has no idea Stata failed and only stopped the build when it couldn't find the output table in the next step.

2) Your process fails (your executable isn't there, or the process nopes out when it gets an error). This is the case for calling Python / calling. `gslab_scons` recognize this as an `ExecCallError`. Since the process returns an error, scons terminates and the log is incomplete.  In that case do this "hack" where we pretend that the log's end time is a fixed date that we know for a fact cannot possibly be. That is so that in the log-printing function we only need to print two kinds of logs: a) the ones that have an end time after this Scons run starts (i.e. of this current run) b) and the ones that broke.